lib/internal/navigator.js

```diff
@@ -27,60 +27,62 @@ const {
 } = internalBinding('os');
 
 const kInitialize = Symbol('kInitialize');
-const nodeVersion = process.version;
+const nodeVersion = require('internal/process/version');
+const platform = require('internal/process/platform');
+const arch = require('internal/process/arch');
 
 /**
  * @param {object} process
  * @param {string} process.platform
  * @param {string} process.arch
  * @returns {string}
  */
-function getNavigatorPlatform(process) {
-  if (process.platform === 'darwin') {
+function getNavigatorPlatform({ platform, arch }) {
+  if (platform === 'darwin') {
     // On macOS, modern browsers return 'MacIntel' even if running on Apple Silicon.
     return 'MacIntel';
-  } else if (process.platform === 'win32') {
+  } else if (platform === 'win32') {
     // On Windows, modern browsers return 'Win32' even if running on a 64-bit version of Windows.
     // https://developer.mozilla.org/en-US/docs/Web/API/Navigator/platform#usage_notes
     return 'Win32';
-  } else if (process.platform === 'linux') {
-    if (process.arch === 'ia32') {
+  } else if (platform === 'linux') {
+    if (arch === 'ia32') {
       return 'Linux i686';
-    } else if (process.arch === 'x64') {
+    } else if (arch === 'x64') {
       return 'Linux x86_64';
     }
-    return `Linux ${process.arch}`;
-  } else if (process.platform === 'freebsd') {
-    if (process.arch === 'ia32') {
+    return `Linux ${arch}`;
+  } else if (platform === 'freebsd') {
+    if (arch === 'ia32') {
       return 'FreeBSD i386';
-    } else if (process.arch === 'x64') {
+    } else if (arch === 'x64') {
       return 'FreeBSD amd64';
     }
-    return `FreeBSD ${process.arch}`;
-  } else if (process.platform === 'openbsd') {
-    if (process.arch === 'ia32') {
+    return `FreeBSD ${arch}`;
+  } else if (platform === 'openbsd') {
+    if (arch === 'ia32') {
       return 'OpenBSD i386';
-    } else if (process.arch === 'x64') {
+    } else if (arch === 'x64') {
       return 'OpenBSD amd64';
     }
-    return `OpenBSD ${process.arch}`;
-  } else if (process.platform === 'sunos') {
-    if (process.arch === 'ia32') {
+    return `OpenBSD ${arch}`;
+  } else if (platform === 'sunos') {
+    if (arch === 'ia32') {
       return 'SunOS i86pc';
     }
-    return `SunOS ${process.arch}`;
-  } else if (process.platform === 'aix') {
+    return `SunOS ${arch}`;
+  } else if (platform === 'aix') {
     return 'AIX';
   }
-  return `${StringPrototypeToUpperCase(process.platform[0])}${StringPrototypeSlice(process.platform, 1)} ${process.arch}`;
+  return `${StringPrototypeToUpperCase(platform[0])}${StringPrototypeSlice(platform, 1)} ${arch}`;
 }
 
 class Navigator {
   // Private properties are used to avoid brand validations.
   #availableParallelism;
   #userAgent;
   #platform;
-  #language;
+  #language = Intl?.Collator().resolvedOptions().locale || 'en-US';
```

Comment:
We can just keep this as a lazy attribute. How does preserving language benefit end users? There shouldn't be a security concern here.  Is there?

Reply:
Since any kind of incorrectness can be viewed as a security concern, I'm not really making a security argument here. However, Navigator is a web/WinterCG standard, and i'm quite certain that in browsers, runtime modifications can't change what `navigator` produces by default, including the language.

Reply:
Yes, but this has an impact on the boot performance with no gain, because there isn't a use case of mutating language that could result in a potential security issue. I think we have to make trade offs.

Reply:
This particular line only executes when navigator is lazy-loaded, I believe, so there shouldn't be any impact on boot performance?

Reply:
which, actually, does mean that I'm not getting any reliability from this change as-is - so it'll have to be cached at boot along with the other three values, or else it should just be a memoizing getter. I'll move it back to that form, for now, and look into caching it at boot.